### ISSUE TITLE
Unexpected/misleading IDE diagnostics for a user defined assignment operator usage

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#78280]
```
class Program
{
    static void Main()
    {
        var c1 = new C1();
        c1++; // IDE0059 Unnecessary assignment of a value to 'c1'

        var c2 = new C1();
        c2 -= 1; // IDE0059 Unnecessary assignment of a value to 'c2'
    }
}

class C1
{
    public void operator ++()
    {
    }

    public void operator -=(int i)
    {
    }
}
```
